src/test/kotlin/com/charleskorn/kaml/YamlWritingTest.kt

```diff
@@ -471,6 +478,118 @@ object YamlWritingTest : Spek({
             }
         }
 
+        describe("handling sealed classes") {
+            val yaml = Yaml(context = sealedModule)
+            context("serializing int sealed class") {
+                val input = SealedWrapper(TestSealedStructure.SimpleSealedInt(5))
+                val output = yaml.stringify(SealedWrapper.serializer(), input)
+
+                it("returns the value serialized in the expected YAML form") {
+                    assert(output).toBe(
+                        """
+                        element: !<sealedInt>
+                          value: 5
+                    """.trimIndent()
+                    )
+                }
+            }
+
+            context("serializing string sealed class") {
+                val input = SealedWrapper(TestSealedStructure.SimpleSealedString("5"))
+                val output = yaml.stringify(SealedWrapper.serializer(), input)
+
+                it("returns the value serialized in the expected YAML form") {
+                    assert(output).toBe(
+                        """
+                        element: !<sealedString>
+                          value: "5"
+                    """.trimIndent()
+                    )
+                }
+            }
+
+            context("serializing list of sealed class structures") {
+                val input = listOf(
+                    TestSealedStructure.SimpleSealedInt(5),
+                    TestSealedStructure.SimpleSealedString("some test"),
+                    TestSealedStructure.SimpleSealedInt(-20),
+                    TestSealedStructure.SimpleSealedString("another test")
+                ).map(::SealedWrapper)
+                val output = yaml.stringify(SealedWrapper.serializer().list, input)
+
+                it("returns the value serialized in the expected YAML form") {
+                    assert(output).toBe(
+                        """
+                        - element: !<sealedInt>
+                            value: 5
+                        - element: !<sealedString>
+                            value: "some test"
+                        - element: !<sealedInt>
+                            value: -20
+                        - element: !<sealedString>
+                            value: "another test"
+                    """.trimIndent()
+                    )
+                }
+            }
+        }
+
+        describe("handling interface") {
```

Comment:
Again, is there anything special we have to do for interfaces? Otherwise I would remove this set of extra tests